src/prefect/engine.py

```diff
@@ -151,6 +151,10 @@
 from prefect.results import BaseResult, ResultFactory
 from prefect.settings import (
     PREFECT_DEBUG_MODE,
+    PREFECT_ENABLE_ON_CANCELLATION_HOOKS,
```

Comment:
two cents: i vote we consolidate these into a single setting, I would be surprised if this level of granularity was meaningful

Reply:
That makes sense. I split them up so the engine could continue to own `on_completion` and `on_failed` hooks. If we consolidated them into one setting, the runner would need to be responsible for executing all the hooks. I think that will be fine, but I mention it in case you think of some scenarios where the runner owning all hooks wouldn't work.

Reply:
I agree that we don't benefit from the runner running the `on_completion` and `on_failed`. It felt strange having a setting to toggle just the `on_cancellation` and `on_crashed` hooks, so I switched to using just an environment variable to avoid introducing any user-facing settings.